fix(ONYX-822): fix broken ArtsyWebView paddings

dariakoko opened the pull request:
This PR resolves [ONYX-822] <!-- eg [PROJECT-XXXX] -->

### Description
<!-- Info, implementation, how to get there, before & after screenshots & videos, follow-up work, etc -->

The ArtsyWebView uses SafeAreaView. SafeAreaView adds paddings to the top and bottom of the page to make sure the active area of the screen can be assessed. We do not need this padding if we are not presenting the screen modally. When we do not the screen modally we can use a react-native View instead of SafeAreaView

| Before iOS | After iOS | Before Android | After Android |
| --- | --- | --- | --- |
| ![image](https://github.com/artsy/eigen/assets/36167539/a4dc33e5-74cc-4bf2-b677-08301140edf7) | ![image](https://github.com/artsy/eigen/assets/36167539/3ca4f7eb-0964-41c5-b356-e8e9522615df) | ![Screenshot_1709343815](https://github.com/artsy/eigen/assets/36167539/58ba30f0-4c7d-4499-9c8a-dffc5eb2095d) | ![Screenshot_1709344209](https://github.com/artsy/eigen/assets/36167539/b6e21408-3197-47b8-b10a-8b2b2be0cfae) |

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

### Changelog updates

<!-- 📝 Please fill out at least one of these sections. -->
<!-- ⓘ 'User-facing' changes will be published as release notes. -->
<!-- ⌫ Feel free to remove sections that don't apply. -->
<!-- • Write a markdown list or just a single paragraph, but stick to plain text. -->
<!-- 📖 eg. `Enable lotsByFollowedArtists - john` or `Fix phone input misalignment - mary`. -->
<!-- 🤷‍♂️ Replace this entire block with the hashtag `#nochangelog` to avoid updating the changelog. -->
<!-- ⚠️ Prefix with `[NEEDS EXTERNAL QA]` if a change requires external QA -->

#### Cross-platform user-facing changes

- in ArtsyWebView use SafeAreaView only if presented modally - daria

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

<!-- end_changelog_updates -->

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md


[ONYX-822]: https://artsyproduct.atlassian.net/browse/ONYX-822?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

ArtsyOpenSource commented:
<!--
  0 failure: 
  0 warning: 
  
  1 markdown notices
  DangerID: danger-id-Danger;
-->



### This PR contains the following changes:

- Cross-platform user-facing changes (in ArtsyWebView use SafeAreaView only if presented modally - daria)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e791ce5c64a0cc2fb73f29379baa6f1d78fee665
</p>

MounirDhahri commented on src/app/routes.ts:
```diff
@@ -279,6 +280,7 @@ function getDomainMap(): Record<string, RouteMatcher[] | null> {
       safeAreaEdges: ["bottom"],
     }),
     addWebViewRoute("/identity-verification-faq"),
+    addWebViewRoute("/galleries"),
```
Nice idea on adding the routes here, but this leads me to think the issue affects any route that is not here. I would therefore look into another approach to get the same behaviour that we get from `addWebViewRoute` to all our deep urls.

olerichter00 replied:
https://github.com/artsy/eigen/pull/9924 would fix the issue by using [ArtsyWebViewPage](https://github.com/artsy/eigen/blob/4a10245a845a6059bebf89a295724268d13ec486/src/app/Components/ArtsyWebView.tsx#L51) for all URLs that down't mach a specific route.

olerichter00 replied:
Please note that my PR would be a fix on top of this PR because this change is still needed: https://github.com/artsy/eigen/pull/9922/files#diff-edc2fa1c2b97a3ddeaf270d241e4054ba768be253b35b4e75e63914b6b1c20cdR257

olerichter00 replied:
It works without adding the routes but it seems that https://github.com/artsy/eigen/pull/9922/files#diff-edc2fa1c2b97a3ddeaf270d241e4054ba768be253b35b4e75e63914b6b1c20cdR257 is needed, no?

MounirDhahri replied:
perfect, do. you mind if you don't commit the route changes then and leave that before merging

olerichter00 replied:
Good idea not to commit the route changes. @dariakoko I'd say, let's merge https://github.com/artsy/eigen/pull/9924 into this PR, then remove the route changes, test again, and then merge into main.

dariakoko replied:
Good idea, doing it now

MounirDhahri reviewed (changes requested):
This is great, I don't think though we should be merging this at its current state since I find this as an opportunity to fix other similar URL routes that we haven't included in `routes.ts` - especially because now we know what is broken

olerichter00 reviewed (commented):
Thanks a lot for the fix 🙏 

Although this fix works with all routes that are added with `addWebViewRoute("/artists")`, routes that are not added like this would be interpreted as vanity URLs (e.g., `/shows`), which results in the Back button being rendered on top of the page title. I've created https://github.com/artsy/eigen/pull/9924 as an attempt to solve this issue as well.

MounirDhahri commented:
We thought about the same thing @olerichter00 -  Looking into your PR now

dariakoko commented:
Addressed your comments @MounirDhahri @olerichter00 and tested - looking good!
![image](https://github.com/artsy/eigen/assets/36167539/0272b34f-89f4-48e0-91c0-8062d0558971)

olerichter00 reviewed (approved):
🌟

MounirDhahri reviewed (approved):
Perfect - our webviews across the app will be much better now!

MounirDhahri commented:
Will merge this to get it to the release
